x-pack/legacy/plugins/reporting/server/routes/generate_from_jobparams.ts

```diff
@@ -96,22 +78,22 @@ export function registerGenerateFromJobParams(
         throw boom.badRequest(`invalid rison: ${jobParamsRison}`);
       }
       try {
-        response = await handler(exportType, jobParams, legacyRequest, h);
+        response = await handler(exportType, jobParams, request, res);
       } catch (err) {
         throw handleError(exportType, err);
       }
       return response;
-    },
-  });
+    })
+  );
 
   // Get route to generation endpoint: show error about GET method to user
-  server.route({
-    path: `${BASE_GENERATE}/{p*}`,
-    method: 'GET',
-    handler: () => {
-      const err = boom.methodNotAllowed('GET is not allowed');
-      err.output.headers.allow = 'POST';
-      return err;
+  router.get(
+    {
+      path: `${BASE_GENERATE}/{p*}`,
+      validate: false,
     },
-  });
+    router.handleLegacyErrors(() => {
+      throw boom.methodNotAllowed('GET is not allowed');
+    })
```

Comment:
You could use `res.customError({statusCode: 405, body: 'GET is not allowed'})` to avoid using both `boom` and `handleLegacyErrors `